App.js

```diff
@@ -1,12 +1,29 @@
-import { StatusBar } from 'expo-status-bar';
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import { StyleSheet, Text, View, Image } from 'react-native';
+import { allPokemons, pokemonByNameOrNumber } from './Data';
+
 
 export default function App() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [dataToDisplay, setDataToDisplay] = useState("");
+  useEffect(() => {
+    pokemonByNameOrNumber(4, setDataToDisplay, setIsLoading, false); 
```

Comment:
I think it's better to have this function to return data, rather than passing state. For example:
```suggestion
    setIsLoading(true);
    pokemonByNameOrNumber(4)
    .then( pokemon => {
        setDataToDisplay(pokemon)
        setIsLoading(false);
    }); 
```